release/nightly_tests/dataset/iter_batches_benchmark.py

```diff
@@ -0,0 +1,99 @@
+import sys
+from typing import Optional
+
+import ray
+from ray.data.dataset import Dataset
+
+from benchmark import Benchmark
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+
+def iter_batches(
+    ds: Dataset,
+    batch_size: Optional[int] = None,
+    batch_format: Literal["default", "pandas", "pyarrow", "numpy"] = "default",
+    local_shuffle_buffer_size: Optional[int] = None,
+    use_default_params: bool = False,
+) -> Dataset:
+    num_batches = 0
+    if use_default_params:
+        for batch in ds.iter_batches():
+            num_batches += 1
+    else:
+        for batch in ds.iter_batches(
+            batch_format=batch_format,
+            batch_size=batch_size,
+            local_shuffle_buffer_size=local_shuffle_buffer_size,
+        ):
+            num_batches += 1
+    print("iter_batches done, num_rows:", ds.count(), "num_batches:", num_batches)
+    return ds
+
+
+def run_iter_batches_benchmark(benchmark: Benchmark):
+    ds = (
+        ray.data.read_parquet(
+            "s3://anonymous@air-example-data/ursa-labs-taxi-data/by_year/2018/01"
+        )
+        .repartition(24)
+        .fully_executed()
+    )
+
+    batch_formats = ["pandas", "numpy"]
+    batch_sizes = [128, 256, 512]
```

Comment:
One point of concern: this is a tabular ingest benchmark, and although these batch sizes are typical/reasonable for CV ingest, I don't know if they are representative for tabular DL.

Some data points:
* A 512 batch size for this NYC taxi dataset is ~40 KB, which is pretty small
* We have tabular DL lighthouse users that are using batch sizes on the order of tens of thousands to hundreds of thousands of records
* Walmart is doing tabular DL with a 16384 batch size: https://resources.nvidia.com/en-us-merlin/gtcs21-s31806?lx=97GH0Q
* TabNet recommends a batch size that's 1-10% of the total dataset size, if memory constraints permit (0.1% to 10% would put us in the 10k - 1M batch size range): https://arxiv.org/pdf/1908.07442.pdf

Reply:
Good point. The setup here is certainly following closely the current default batch_size (256) for this param (which I think we should comment in code and call out it's good for image but not necessarily for tabular). I'm pushing up this number and it's significantly faster (100min to iter through one file before was indeed too slow...).

Reply:
Maybe a good set of batch sizes would be something like `[4 * 1024, 64 * 1024, 512 * 1024]` or something like that. 🤔

Reply:
Yeah, spaced them out by 4x now, starting from 4*1024.